hnn_core/feed.py

```diff
@@ -35,252 +74,238 @@ class ExtFeed(object):
+    """Create gaussian input.
+
+    Parameters
+    ----------
+    mu : float
+        The mean time of spikes.
+    sigma : float
+        The standard deviation.
+
+    Returns
+    -------
+    event_times : array
+        The event times.
+    """
+    event_times = prng.normal(mu, sigma, 50)
+    return event_times
+
+
+def _create_common_input(distribution, t0, t0_stdev, tstop, f_input,
+                         stdev, repeats, events_per_cycle, prng, prng2):
+    """Creates the common ongoing external inputs.
+
+    Used for, e.g., for rhythmic inputs in alpha/beta generation.
+
+    Parameters
+    ----------
+    distribution : str
+        The distribution for each burst. One of 'normal' or 'uniform'.
+    t0 : float
+        The start times. If -1, then randomize the start time
+        of inputs.
+    t0_stdev : float
+        Standard deviation of jitter to start time.
+    tstop : float
+        The stop time.
+    f_input : float
+        The frequency of input bursts.
+    stdev : float
+        The standard deviation.
+    repeats : int
+        The number of repeats.
+    events_per_cycle : float
+        The events per cycle. Must be 1 or 2.
+    prng : instance of RandomState
+        The random state.
+    prng : instance of RandomState
+        The random state used for jitter to start time (see t0_stdev).
+
+    Returns
+    -------
+    event_times : array
+        The event times.
+    """
+    if distribution not in ('normal', 'uniform'):
+        raise ValueError("Indicated distribution not recognized. "
+                         "Not making any common feeds.")
+
+    # store f_input as self variable for later use if it exists in p
+    if t0 == -1:
+        t0 = prng.uniform(25., 125.)
+    elif t0_stdev > 0.0:
+        t0 = prng2.normal(t0, t0_stdev)
+
+    if events_per_cycle != 1:
+        print("events_per_cycle should be either 1 or 2, trying 2")
+        events_per_cycle = 2
+
+    if distribution == 'normal':
+        # array of mean stimulus times, starts at t0
+        isi_array = np.arange(t0, tstop, 1000. / f_input)
+        # array of single stimulus times -- no doublets
+        if stdev > 0:
+            t_array = prng.normal(np.repeat(isi_array, repeats), stdev)
         else:
-            print("Indicated distribution not recognized. "
-                  "Not making any common feeds.")
-            t_input = np.array([])
-        self.event_times = t_input.tolist()
+            t_array = isi_array
+    elif distribution == 'uniform':
+        n_inputs = repeats * f_input * (tstop - t0) / 1000.
+        t_array = prng.uniform(t0, tstop, n_inputs)
+
+    t_input = np.array([])
+    if events_per_cycle == 2:
+        # Two arrays store doublet times
+        t_input_low = t_array - 5
+        t_input_high = t_array + 5
+        # Array with ALL stimulus times for input
+        # np.append concatenates two np arrays
+        t_input = np.append(t_input_low, t_input_high)
```

Comment:
So this isn't a comment on this PR, more to the code: Could we get one of the OPs to give us a link to the part of documentation where this is specified?

Reply:
maybe @rythorpe or @ntolley can help since they use the GUI?

Reply:
I don't understand. What specifically are you looking for here @cjayb?

Reply:
The code is now simplified to

```python
return np.append(t_array - 5, t_array + 5)
```

but doesn't it creep you out that there are undocumented, hard-coded time shifts there? Clearly it's creating spike "doublets", and certainly for a very specific reason. I was just wondering whether anyone still remembers what the reason was?

Reply:
> but doesn't it creep you out that there are undocumented, hard-coded time shifts there?

LOL yes :) Maybe it makes sense for some reason, I don't know ...

I think @rythorpe or @ntolley can throw some light on what this function does and how the spike train is supposed to look like and why. I opened the GUI many aeons ago ...

Reply:
Perhaps you should restore the comment explaining how this is equivalent to the spikes/burst parameter in GUI?. As for the fixed time shifts, I'm guessing there is a biologic justification for a doublet with a 10 ms ISI. 

I see that it was added in a pretty large commit in the [GUI](https://github.com/jonescompneurolab/hnn/commit/2161e2c75bf0378080848093b987f7424b1c6f85#diff-84da9608764aa67ace81318363636bd272c6053c686b2be6616187c04325bcfdR170).

Reply:
I think the 10 ms is a somewhat standard approximation for the amount of time between spikes within a thalamic burst. The [HNN webset/tutorials](https://jonescompneurolab.github.io/hnn-under_the_hood/04_evoked-rhythmic-inputs/04_evoked-rhythmic-inputs) don't explicitly address this, however, they do reference [this](https://pubmed.ncbi.nlm.nih.gov/16061522/) paper that discusses the bursting properties of thalamic drive into cortex.

Reply:
Ideally we'll define an API in a later PR for creating rhythmic feeds that allows the user to define the intra-burst ISI.

Reply:
done, I updated the docstring :) A demo of the GUI for these functions would be nice one day, now that we have everything nicely documented in the code.